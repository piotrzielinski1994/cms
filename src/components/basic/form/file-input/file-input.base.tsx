import { HtmlProps } from '@/utils/html/html.types';
import { forwardRef } from 'react';
import Form from '../root/form';

type InputProps = {
  name: string;
  value?: string;
  accept?: string;
  multiple?: boolean;
} & Omit<HtmlProps['input'], 'type' | 'value' | 'accept' | 'multiple'>;

const Wrapper = (props: HtmlProps['label']) => {
  return <label {...props} />;
};

const Input = forwardRef<HTMLInputElement, InputProps>((props, ref) => {
  return <input type="file" ref={ref} {...props} />;
});

const Items = (props: HtmlProps['ul']) => {
  return <ul {...props} />;
};

const Item = (props: HtmlProps['li']) => {
  return <li {...props} />;
};

const CloseAllButton = (props: HtmlProps['button']) => {
  return <button type="button" {...props} />;
};

Input.displayName = 'FileInputBase.Input';

const FileInputBase = {
  Root: Form.Group,
  Label: Form.Label,
  Wrapper,
  Input,
  Items,
  Item,
  CloseAllButton,
  Error: Form.Error,
};

export default FileInputBase;
