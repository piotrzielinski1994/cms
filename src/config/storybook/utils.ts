import { ReactNode } from 'react';

const DEFAULT_VALUE = '__default__' as const;
const THUMBNAIL_ID = 'thumbnail';

const storyToUrlPath = (title: string) => {
  const parts = title.split('/');
  const story = parts[parts.length - 1];
  const titlePart = title.toLowerCase().replace(/\//g, '-').replace(/\s+/g, '-');
  const storyPart = story
    .replace(/([a-z0-9])([A-Z])/g, '$1-$2')
    .replace(/([a-zA-Z])(\d+)/g, '$1-$2')
    .toLowerCase();
  return `${titlePart}--${storyPart}`;
};

const getFallback = <T extends ReactNode>(value: T, translation: string): T => {
  return value === DEFAULT_VALUE ? (translation as T) : value;
};

export { DEFAULT_VALUE, getFallback, storyToUrlPath, THUMBNAIL_ID };
