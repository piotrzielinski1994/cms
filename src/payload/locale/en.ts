const en = {
  common: {
    content: 'Content',
    layout: 'Layout',
    component: {
      singular: 'Component',
      plural: 'Components',
    },
    section: {
      singular: 'Section',
      plural: 'Sections',
    },
    related: {
      singular: 'Related',
      plural: 'Related',
    },
    openInNewTab: 'Open in new tab',
    item: {
      singular: 'Item',
      plural: 'Items',
    },
    files: {
      singular: 'File',
      plural: 'Files',
    },
    default: 'Default',
    dark: 'Dark',
    question: 'Question',
    answer: 'Answer',
    pageNotFound: 'This page could not be found.',
    goHome: 'Go home',
  },
  fields: {
    label: 'Label',
    slug: 'Slug',
    path: 'Path',
    type: 'Type',
    title: 'Title',
    heading: 'Heading',
    subheading: 'Subheading',
    description: 'Description',
    image: 'Image',
    publishedAt: 'Published at',
    fullName: 'Full name',
    parent: 'Parent element',
    children: 'Child elements',
    role: {
      singular: 'Role',
      plural: 'Roles',
    },
    menu: 'Menu',
    submenu: 'Submenu',
    navItem: 'Navigation item',
    internalLink: 'Internal link',
    documentToLinkTo: 'Document to link to',
    buttons: 'Buttons',
    selector: 'Selector',
    isReversed: 'Reversed layout',
    subpages: 'Subpages',
    seo: 'SEO',
    acceptLabel: 'Accept label',
    readMoreLabel: 'Read more label',
    url: 'URL',
  },
  groups: {
    admin: 'Admin',
    docs: 'Docs',
  },
  collections: {
    pages: {
      singular: 'Page',
      plural: 'Pages',
    },
    categories: {
      singular: 'Category',
      plural: 'Categories',
    },
    posts: {
      singular: 'Post',
      plural: 'Posts',
    },
    sites: {
      singular: 'Site',
      plural: 'Sites',
    },
    users: {
      singular: 'User',
      plural: 'Users',
    },
    images: {
      singular: 'Image',
      plural: 'Images',
    },
    redirects: {
      singular: 'Redirect',
      plural: 'Redirects',
    },
  },
  enums: {
    role: {
      user: 'User',
      editor: 'Editor',
      admin: 'Admin',
    },
  },
  components: {
    header: 'Header',
    footer: 'Footer',
    cookiesBanner: 'Cookies banner',
  },
  frontend: {
    submit: 'Submit',
    logo: 'Logo',
    localeSwitcher: 'Locale switcher',
    themeSwitcher: 'Theme switcher',
    fontScaleSwitcher: 'Font scale switcher',
    increment: 'Increment',
    decrement: 'Decrement',
    close: 'Close',
    component: {
      skipLink: 'Go to the main content',
    },
    contactForm: {
      fields: {
        email: {
          label: 'Email',
        },
        message: {
          label: 'Message',
        },
      },
    },
  },
  zod: {
    invalid_type: 'Expected {expected}, received {received}',
    invalid_type_with_path: '{path} is expected {expected}, but received {received}',
    invalid_type_received_undefined: 'Required',
    invalid_literal: 'Invalid literal value, expected {expected}',
    unrecognized_keys: 'Unrecognized key(s) in object: {- keys}',
    unrecognized_keys_one: 'Unrecognized key in object: {- keys}',
    unrecognized_keys_other: 'Unrecognized keys in object: {- keys}',
    invalid_union: 'Invalid input',
    invalid_union_discriminator: 'Invalid discriminator value. Expected {- options}',
    invalid_enum_value: 'Invalid enum value. Expected {- options}, received {received}',
    invalid_arguments: 'Invalid function arguments',
    invalid_return_type: 'Invalid function return type',
    invalid_date: 'Invalid date',
    custom: 'Invalid input',
    invalid_intersection_types: 'Intersection results could not be merged',
    not_multiple_of: 'Number must be a multiple of {multipleOf}',
    not_finite: 'Number must be finite',
    invalid_string: {
      email: 'Invalid {validation}',
      url: 'Invalid {validation}',
      uuid: 'Invalid {validation}',
      cuid: 'Invalid {validation}',
      regex: 'Invalid',
      datetime: 'Invalid {validation}',
      startsWith: 'Invalid input: must start with {startsWith}',
      endsWith: 'Invalid input: must end with {endsWith}',
    },
    too_small: {
      array: {
        inclusive: 'Array must contain at least {minimum} element(s)',
        inclusive_one: 'Array must contain at least {minimum} element',
        inclusive_other: 'Array must contain at least {minimum} elements',
        not_inclusive: 'Array must contain more than {minimum} element(s)',
        not_inclusive_one: 'Array must contain more than {minimum} element',
        not_inclusive_other: 'Array must contain more than {minimum} elements',
      },
      string: {
        inclusive: 'String must contain at least {minimum} character(s)',
        inclusive_one: 'String must contain at least {minimum} character',
        inclusive_other: 'String must contain at least {minimum} characters',
        inclusive_with_path: '{path} must contain at least {minimum} character(s)',
        inclusive_with_path_one: '{path} must contain at least {minimum} character',
        inclusive_with_path_other: '{path} must contain at least {minimum} characters',
        not_inclusive: 'String must contain over {minimum} character(s)',
        not_inclusive_one: 'String must contain over {minimum} character',
        not_inclusive_other: 'String must contain over {minimum} characters',
        not_inclusive_with_path: '{path} must contain over {minimum} character(s)',
        not_inclusive_with_path_one: '{path} must contain over {minimum} character',
        not_inclusive_with_path_other: '{path} must contain over {minimum} characters',
      },
      number: {
        inclusive: 'Number must be greater than or equal to {minimum}',
        inclusive_with_path: '{path} must be greater than or equal to {minimum}',
        not_inclusive: 'Number must be greater than {minimum}',
        not_inclusive_with_path: '{path} must be greater than {minimum}',
      },
      set: {
        inclusive: 'Invalid input',
        not_inclusive: 'Invalid input',
      },
      date: {
        inclusive: 'Date must be greater than or equal to {- minimum, datetime}',
        not_inclusive: 'Date must be greater than {- minimum, datetime}',
      },
    },
    too_big: {
      array: {
        inclusive: 'Array must contain at most {maximum} element(s)',
        inclusive_one: 'Array must contain at most {maximum} element',
        inclusive_other: 'Array must contain at most {maximum} elements',
        not_inclusive: 'Array must contain less than {maximum} element(s)',
        not_inclusive_one: 'Array must contain less than {maximum} element',
        not_inclusive_other: 'Array must contain less than {maximum} elements',
      },
      string: {
        inclusive: 'String must contain at most {maximum} character(s)',
        inclusive_one: 'String must contain at most {maximum} character',
        inclusive_other: 'String must contain at most {maximum} characters',
        inclusive_with_path: '{path} must contain at most {maximum} character(s)',
        inclusive_with_path_one: '{path} must contain at most {maximum} character',
        inclusive_with_path_other: '{path} must contain at most {maximum} characters',
        not_inclusive: 'String must contain under {maximum} character(s)',
        not_inclusive_one: 'String must contain under {maximum} character',
        not_inclusive_other: 'String must contain under {maximum} characters',
        not_inclusive_with_path: '{path} must contain under {maximum} character(s)',
        not_inclusive_with_path_one: '{path} must contain under {maximum} character',
        not_inclusive_with_path_other: '{path} must contain under {maximum} characters',
      },
      number: {
        inclusive: 'Number must be less than or equal to {maximum}',
        inclusive_with_path: '{path} must be less than or equal to {maximum}',
        not_inclusive: 'Number must be less than {maximum}',
        not_inclusive_with_path: '{path} must be less than {maximum}',
      },
      set: {
        inclusive: 'Invalid input',
        not_inclusive: 'Invalid input',
      },
      date: {
        inclusive: 'Date must be smaller than or equal to {- maximum, datetime}',
        not_inclusive: 'Date must be smaller than {- maximum, datetime}',
      },
    },
  },
  storybook: {
    basic: {
      form: {
        textInput: {
          default: {
            label: 'Text Input',
            placeholder: 'Type value',
          },
          disabled: {
            label: 'Disabled Input',
            placeholder: 'Disabled',
          },
          invalid: {
            label: 'Invalid Input',
            value: 'Invalid value',
            error: 'Error message',
          },
        },
        numberInput: 'Number Input',
        textArea: 'Text Area',
        checkbox: {
          default: {
            label: 'Checkbox',
          },
          disabled: {
            label: 'Disabled Checkbox',
          },
          invalid: {
            label: 'Invalid Checkbox',
            error: 'Error message',
          },
        },
        radio: {
          default: {
            label: 'Radio Button',
          },
          disabled: {
            label: 'Disabled Radio Button',
          },
          invalid: {
            label: 'Invalid Radio Button',
            error: 'Error message',
          },
        },
        select: {
          label: 'Select',
          placeholder: 'Choose value',
          option: 'Option',
        },
      },
      button: {
        default: 'Button',
        disabled: 'Disabled',
        secondary: 'Secondary',
      },
      alert: {
        info: 'Information',
        success: 'Success',
        warn: 'Warning',
        error: 'Error',
      },
      badge: 'Badge',
      accordion: {
        heading: 'Heading',
        content: 'Content',
      },
      dialog: {
        button: 'Show Dialog',
        header: 'Header',
        content: 'Content',
        submit: 'Submit',
        cancel: 'Cancel',
      },
    },
  },
};

export { en };
