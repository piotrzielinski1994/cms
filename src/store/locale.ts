import { contentLocales, defaultContentLocale } from '@/config/locales.config';
import { Locale, useLocale } from 'next-intl';

const useLocaleStore = (): Locale => {
  const locale = useLocale();
  return contentLocales.includes(locale) ? locale : defaultContentLocale;
};

export { useLocaleStore };
