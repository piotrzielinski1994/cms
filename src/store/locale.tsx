'use client';

import { contentLocales, defaultContentLocale, translations } from '@/config/store/locales.config';
import { Locale, NextIntlClientProvider, useLocale } from 'next-intl';
import { PropsWithChildren, useEffect } from 'react';
import z from 'zod';

const useLocaleStore = (): Locale => {
  const locale = useLocale();
  return contentLocales.includes(locale) ? locale : defaultContentLocale;
};

const LocaleProvider = ({ children, locale }: PropsWithChildren & { locale: Locale }) => {
  useEffect(() => {
    z.config(z.locales[locale]());
  }, [locale]);

  return (
    <NextIntlClientProvider locale={locale} messages={translations[locale]}>
      {children}
    </NextIntlClientProvider>
  );
};

export { LocaleProvider, useLocaleStore };
