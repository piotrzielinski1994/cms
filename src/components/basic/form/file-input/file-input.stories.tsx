import Form from '@/components/basic/form/root/form';
import { DEFAULT_VALUE, getFallback } from '@/config/storybook/utils';
import type { Meta, StoryObj } from '@storybook/react';
import { useTranslations } from 'next-intl';
import { useState, type ComponentProps } from 'react';
import { FileInput as FileInputComponent } from './file-input';

type Args = ComponentProps<typeof FileInputComponent> & {
  label: string;
};

const meta: Meta<Args> = {
  component: FileInputComponent,
  title: 'Basic/Form/FileInput',
  argTypes: {
    label: { control: 'text' },
    multiple: { control: 'boolean' },
    disabled: { control: 'boolean' },
    placeholder: { control: 'text' },
    fileNames: { table: { disable: true } },
    onClear: { table: { disable: true } },
  },
  args: {
    label: DEFAULT_VALUE,
    multiple: true,
    disabled: false,
    placeholder: DEFAULT_VALUE,
  },
};

const Render = ({ label, placeholder, ...args }: Args) => {
  const t = useTranslations('storybook.basic.form');
  const [fileNames, setFileNames] = useState<string[]>([
    t('fileInput.file', { index: 1 }),
    t('fileInput.file', { index: 2 }),
  ]);

  return (
    <div className="grid gap-2">
      <Form.Group>
        <Form.Label htmlFor="inpu1">{getFallback(label, t('fileInput.label'))}</Form.Label>
        <FileInputComponent
          {...args}
          placeholder={getFallback(placeholder, t('fileInput.placeholder'))}
          id="inpu1"
          name="inpu1"
          fileNames={fileNames}
          onChange={(e) => {
            const { files } = e.target;
            if (!files) return;
            setFileNames([...files].map((file) => file.name));
          }}
          onClear={() => setFileNames([])}
        />
      </Form.Group>
      <Form.Group>
        <Form.Label htmlFor="input2">{t('textInput.disabled.label')}</Form.Label>
        <FileInputComponent
          id="inpu2"
          name="inpu2"
          placeholder={t('fileInput.placeholder')}
          fileNames={[]}
          disabled
        />
      </Form.Group>
      <Form.Group>
        <Form.Label htmlFor="input3">{t('textInput.invalid.label')}</Form.Label>
        <FileInputComponent
          id="inpu3"
          name="inpu3"
          placeholder={t('fileInput.placeholder')}
          fileNames={[]}
          error={t('textInput.invalid.error')}
        />
      </Form.Group>
    </div>
  );
};

const FileInput: StoryObj<typeof FileInputComponent> = { render: Render };

export { FileInput };
export default meta;
