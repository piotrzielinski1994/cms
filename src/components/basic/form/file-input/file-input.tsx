'use client';

import { cn } from '@/utils/tailwind';
import { Upload, X } from 'lucide-react';
import { useTranslations } from 'next-intl';
import { ChangeEvent, ComponentProps, forwardRef, ReactNode, useState } from 'react';
import { inputClassNames } from '../text-input/text-input';
import FileInputBase from './file-input.base';

type FileInputProps = ComponentProps<typeof FileInputBase.Input> & {
  label?: ReactNode;
  error?: string;
  fileNames: string[];
  onClear?: () => void;
};

const classNames = {
  wrapper: ({ isValid, isDragging }: { isValid: boolean; isDragging: boolean }) =>
    cn(
      inputClassNames.input({ isValid }),
      'has-[:enabled]:hover:border-foreground/90 has-[:enabled]:hover:ring-foreground/90',
      { 'has-[:enabled]:ring-1': isDragging },
      'flex items-start gap-2',
      'cursor-pointer has-[:disabled]:cursor-not-allowed',
    ),
};

const Component = forwardRef<HTMLInputElement, FileInputProps>((props, ref) => {
  const { fileNames, onClear, label, error, ...rest } = props;
  const t = useTranslations('frontend');
  const [isDragging, setIsDragging] = useState(false);

  return (
    <Root>
      {label && <Label htmlFor={rest.id}>{label}</Label>}
      <Wrapper
        isDragging={isDragging}
        error={error}
        aria-label={t('component.uploadInput.clickToUpload')}
        onDragLeave={() => setIsDragging(false)}
        onDragOver={(e) => {
          e.preventDefault();
          setIsDragging(true);
        }}
        onDrop={(e) => {
          e.preventDefault();
          setIsDragging(false);
          const { files } = e.dataTransfer;
          const target = Object.assign(e.target, { files });
          const currentTarget = Object.assign(e.currentTarget, { files });
          const newEvent = Object.assign({}, e, { target, currentTarget });
          props.onChange?.(newEvent as unknown as ChangeEvent<HTMLInputElement>);
        }}
      >
        <Input ref={ref} {...rest} />
        <Upload />
        {fileNames.length > 0 ? (
          <>
            <Items>
              {fileNames.map((fileName, index) => (
                <Item key={index}>{fileName}</Item>
              ))}
            </Items>
            <CloseAllButton onClick={onClear} aria-label={t('close')} />
          </>
        ) : (
          <Placeholder>{props.placeholder}</Placeholder>
        )}
      </Wrapper>
      <Error>{error}</Error>
    </Root>
  );
});

const Wrapper = (
  props: ComponentProps<typeof FileInputBase.Wrapper> &
    Pick<FileInputProps, 'error'> & { isDragging: boolean },
) => {
  const { isDragging, error, className, ...rest } = props;
  const defaultClassNames = cn(classNames.wrapper({ isValid: !error, isDragging }), className);
  return <FileInputBase.Wrapper className={defaultClassNames} {...rest} />;
};

const Input = forwardRef<HTMLInputElement, ComponentProps<typeof FileInputBase.Input>>(
  ({ className, ...rest }, ref) => {
    return <FileInputBase.Input ref={ref} className={cn('sr-only', className)} {...rest} />;
  },
);

const Items = ({ className, ...rest }: ComponentProps<typeof FileInputBase.Items>) => {
  const defaultClassNames = cn('flex-grow grid gap-1', className);
  return <FileInputBase.Items className={defaultClassNames} {...rest} />;
};

const CloseAllButton = (props: ComponentProps<typeof FileInputBase.CloseAllButton>) => {
  return (
    <FileInputBase.CloseAllButton {...props}>
      <X height="1lh" width="auto" />
    </FileInputBase.CloseAllButton>
  );
};

const Placeholder = ({ className, ...rest }: ComponentProps<typeof FileInputBase.Placeholder>) => {
  const defaultClassNames = cn('flex-grow text-foreground/50', className);
  return <FileInputBase.Placeholder className={defaultClassNames} {...rest} />;
};

const Root = FileInputBase.Root;
const Label = FileInputBase.Label;
const Item = FileInputBase.Item;
const Error = FileInputBase.Error;

Component.displayName = 'FileInputBase';
Input.displayName = 'FileInputBase.Input';

const FileInput = Object.assign(Component, {
  Root,
  Label,
  Input,
  Items,
  Item,
  CloseAllButton,
  Placeholder,
  Error,
});

export { FileInput };
