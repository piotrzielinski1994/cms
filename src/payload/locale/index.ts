import { clientEnv } from '@/env.client';
import {
  DefaultTranslationKeys,
  NestedKeysStripped,
  SupportedLanguages,
  TFunction,
} from '@payloadcms/translations';
import { en } from '@payloadcms/translations/languages/en';
import { pl } from '@payloadcms/translations/languages/pl';
import { pick } from 'ramda';
import { en as customEn } from './en';
import { pl as customPl } from './pl';

// Types ====================================
export type AdminTranslations = TFunction<
  NestedKeysStripped<typeof customEn> | DefaultTranslationKeys
>;

// Variables ====================================
export const adminLocale = {
  list: pick(clientEnv.feature.locale.admin.list, { en, pl } as SupportedLanguages),
  customList: {
    en: customEn,
    pl: customPl,
  },
  default: clientEnv.feature.locale.admin.default,
};

export const contentLocale = {
  list: clientEnv.feature.locale.content.list,
  default: clientEnv.feature.locale.content.default,
};
