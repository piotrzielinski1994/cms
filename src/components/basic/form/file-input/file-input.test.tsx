import { en } from '@/payload/locale/en';
import { withProviders } from '@/utils/tests';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ComponentProps } from 'react';
import { describe, expect, it, vi } from 'vitest';
import { axe } from 'vitest-axe';
import { FileInput } from './file-input';

const t = en.frontend;

describe('FileInput', () => {
  const defaultProps = {
    name: 'file-input',
    fileNames: ['file-1.webp', 'file-2.webp'],
    onClear: vi.fn(),
    onChange: vi.fn(),
  } satisfies ComponentProps<typeof FileInput>;

  it('should have no accessibility violations', async () => {
    const { container } = render(withProviders(<FileInput {...defaultProps} />));
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it('should match the snapshot', () => {
    const { container } = render(withProviders(<FileInput {...defaultProps} />));
    expect(container).toMatchSnapshot();
  });

  describe('Interactivity', () => {
    it('should call onChange when file is dropped', async () => {
      const newFileName = 'file-3.webp';
      const { getByLabelText } = render(withProviders(<FileInput {...defaultProps} />));

      const label = getByLabelText(t.component.uploadInput.clickToUpload);
      const input = label.querySelector('input') as HTMLInputElement;
      const file = new File(['file contents'], newFileName, { type: 'image/webp' });

      await userEvent.upload(input, file);

      const dataTransfer = new DataTransfer();
      dataTransfer.items.add(file);

      await userEvent.pointer([{ target: label, keys: '[DragEnter]' }]);
      label.dispatchEvent(
        new DragEvent('drop', {
          bubbles: true,
          cancelable: true,
          dataTransfer,
        }),
      );

      const calledEvent = defaultProps.onChange.mock.calls[0][0];
      expect(calledEvent.target.files![0].name).toBe(newFileName);
      expect(defaultProps.onChange).toHaveBeenCalled();
    });

    it('should call onClear when close button is clicked', async () => {
      const { getByRole } = render(withProviders(<FileInput {...defaultProps} />));
      const button = getByRole('button', { name: t.close });

      await userEvent.click(button);

      expect(defaultProps.onClear).toHaveBeenCalled();
    });
  });

  describe('Keyboard', () => {
    it('should move focus with arrow keys', async () => {
      const { getByRole, getByLabelText } = render(withProviders(<FileInput {...defaultProps} />));
      const input = getByLabelText(t.component.uploadInput.clickToUpload).querySelector('input');

      // Focus input
      await userEvent.tab();
      expect(document.activeElement).toBe(input);

      // Focus clear button
      await userEvent.tab();
      expect(document.activeElement).toBe(getByRole('button', { name: t.close }));

      // Move out of component
      await userEvent.tab();
      expect(document.activeElement).toBe(document.body);
    });
  });
});
