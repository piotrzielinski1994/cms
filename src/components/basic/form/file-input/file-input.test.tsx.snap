// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FileInput > should match the snapshot 1`] = `
<div>
  <div
    class="grid gap-1 content-start grid-rows-[auto_1fr]"
  >
    <label
      aria-invalid="false"
      aria-label="Click to upload"
      class="p-2 border border-solid border-current bg-input placeholder-foreground/50 [&:enabled]:hover:border-foreground/90 [&:enabled]:hover:ring-foreground/90 disabled:cursor-not-allowed disabled:text-foreground/50 outline-none ring-inset focus-within:ring-1 ring-current has-[:enabled]:hover:border-foreground/90 has-[:enabled]:hover:ring-foreground/90 has-[:disabled]:text-foreground/50 flex items-start gap-2 cursor-pointer has-[:disabled]:cursor-not-allowed"
    >
      <input
        class="sr-only"
        name="file-input"
        type="file"
      />
      <svg
        aria-hidden="true"
        class="lucide lucide-upload"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 3v12"
        />
        <path
          d="m17 8-5-5-5 5"
        />
        <path
          d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
        />
      </svg>
      <ul
        class="flex-grow grid gap-1"
      >
        <li>
          file-1.webp
        </li>
        <li>
          file-2.webp
        </li>
      </ul>
      <button
        aria-label="Close"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-x"
          fill="none"
          height="1lh"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="auto"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 6 6 18"
          />
          <path
            d="m6 6 12 12"
          />
        </svg>
      </button>
    </label>
    <span
      aria-hidden="true"
      class="min-h-[1em] text-sm text-red-500 leading-none"
      role="alert"
    />
  </div>
</div>
`;
